src/basic/virt.c

```diff
@@ -273,8 +273,8 @@ static Virtualization detect_vm_dmi(void) {
                                                 " assuming virtualized: %m");
                                 return VIRTUALIZATION_AMAZON;
                         }
-                        if (endswith(truncate_nl(s), ".metal")) {
-                                log_debug("DMI product name ends with '.metal', assuming no virtualization");
+                        if (strstr(s, ".metal")) {
```

Comment:
wouldn't a better check be something like this:

```c
e = strstr(truncate_nl(s), ".metal");
if (e && IN_SET(e[STRLEN(".metal")], 0, '-')) {
        …
```
or in other words, as per your example, we wouldn't match on ".metallica", but only match if .metal is suffix or followed by a "-" separator.

Reply:
now, since #27270 has been merged:

```c
e = strstrafter(runcate_nl(s), ".metal"));
if (e && IN_SET(*e, 0, '-')) {
        …
```

Reply:
I thought about it but I am not entirely certain it is necessary and I worry about yet another change in the instance naming scheme (ie. `.metal.something` for example). I can if you insist, it would work today.

Any reason why the `truncate_nl()` is useful in this case ? `strstr` will find the string wether there's a newline or not...

Reply:
Actually ignore the above, EC2 folks confirmed that syntax is going to work for the foreseable future so I'll do as you  suggest & respin. Thanks !